seyren-core/src/main/java/com/seyren/core/service/notification/OpsGenieNotificationService.java

```diff
@@ -42,70 +45,87 @@
             }
-		} catch (OpsGenieClientException e) {
-			LOGGER.warn(
-					"Did not send notification to OpsGenie for check in state: {}",
-					check.getState(), e);
-		} catch (IOException e) {
-			LOGGER.warn(
-					"Did not send notification to OpsGenie for check in state: {}",
-					check.getState(), e);
-		} catch (ParseException e) {
-			LOGGER.warn(
-					"Did not send notification to OpsGenie for check in state: {}",
-					check.getState(), e);
+        } catch (OpsGenieClientException e) {
+            LOGGER.warn(
+                "Did not send notification to OpsGenie for check in state: {}",
+                check.getState(), e);
+        } catch (IOException e) {
+            LOGGER.warn(
+                "Did not send notification to OpsGenie for check in state: {}",
+                check.getState(), e);
+        } catch (ParseException e) {
+            LOGGER.warn(
+                "Did not send notification to OpsGenie for check in state: {}",
+                check.getState(), e);
+        }
+    }
 
-		}
+    private List<com.ifountain.opsgenie.client.model.beans.Alert> getOpenAlerts(OpsGenieClient client, String apiKey, Check check) throws OpsGenieClientException, IOException, ParseException {
+        List<com.ifountain.opsgenie.client.model.beans.Alert> openMatchingAlerts = new ArrayList<com.ifountain.opsgenie.client.model.beans.Alert>();
+        ListAlertsRequest request = new ListAlertsRequest();
+        request.setApiKey(apiKey);
+        ListAlertsResponse response = client.alert().listAlerts(request);
+        List<com.ifountain.opsgenie.client.model.beans.Alert> alerts = response.getAlerts();
+        //There could be multiple open alerts for a given check
+        for (com.ifountain.opsgenie.client.model.beans.Alert alert: alerts) {
+            if (String.valueOf(alert.getMessage()).equals(getMessage(check)) &&
+                alert.getStatus() == com.ifountain.opsgenie.client.model.beans.Alert.Status.open) {
+                openMatchingAlerts.add(alert);
+            }
+        }
+        return openMatchingAlerts;
     }
 
-	private com.ifountain.opsgenie.client.model.beans.Alert findAlert(OpsGenieClient client, String apiKey, String message) throws OpsGenieClientException, IOException, ParseException {
-		ListAlertsRequest request = new ListAlertsRequest();
-		request.setApiKey(apiKey);
+    private void closeOpenAlerts(OpsGenieClient client, String apiKey, Check check) throws OpsGenieClientException, ParseException, IOException{
+        List<com.ifountain.opsgenie.client.model.beans.Alert> opsAlerts = getOpenAlerts(client, apiKey, check);
+        for(com.ifountain.opsgenie.client.model.beans.Alert alert : opsAlerts) {
+            CloseAlertRequest request = new CloseAlertRequest();
+            request.setId(alert.getId());
+            request.setApiKey(apiKey);
+            CloseAlertResponse response = client.alert().closeAlert(request);
+            assert response.isSuccess();
```

Comment:
Assert statements are funny. They will not be enforced by default - the JVM has to be started with a specific flag to start enforcing them. I know it's just cut'n'paste, but if the intent here is to throw an error if there's a response error, then this won't do it, unless we're starting Seyren with -enableassertions or -ea flags.

Assuming you do want an exception here, this will quit out on the first error. Is that what you want? Or do you want to keep processing the other records?